SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorCommandLineArgsManager.cpp

```diff
@@ -0,0 +1,280 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialGDKEditorCommandLineArgsManager.h"
+
+#include "HAL/PlatformFilemanager.h"
+#include "IOSRuntimeSettings.h"
+#include "Misc/App.h"
+#include "Misc/FileHelper.h"
+#include "Misc/MessageDialog.h"
+#include "Serialization/JsonSerializer.h"
+
+#ifdef ENABLE_LAUNCHER_DELEGATE
+#include "ILauncher.h"
+#include "ILauncherServicesModule.h"
+#include "ILauncherWorker.h"
+#endif // ENABLE_LAUNCHER_DELEGATE
+
+#include "SpatialCommandUtils.h"
+#include "SpatialGDKEditorSettings.h"
+#include "SpatialGDKSettings.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialGDKEditorCommandLineArgsManager);
+
+FSpatialGDKEditorCommandLineArgsManager::FSpatialGDKEditorCommandLineArgsManager()
+#ifdef ENABLE_LAUNCHER_DELEGATE
+	: bAndroidDevice(false)
+#endif // ENABLE_LAUNCHER_DELEGATE
+{
+}
+
+void FSpatialGDKEditorCommandLineArgsManager::Init()
+{
+#ifdef ENABLE_LAUNCHER_DELEGATE
+	ILauncherServicesModule& LauncherServicesModule = FModuleManager::LoadModuleChecked<ILauncherServicesModule>(TEXT("LauncherServices"));
+	LauncherServicesModule.OnCreateLauncherDelegate.AddRaw(this, &FSpatialGDKEditorCommandLineArgsManager::OnCreateLauncher);
+#endif // ENABLE_LAUNCHER_DELEGATE
+}
+
+#ifdef ENABLE_LAUNCHER_DELEGATE
+void FSpatialGDKEditorCommandLineArgsManager::OnLauncherCanceled(double ExecutionTime)
+{
+	RemoveCommandLineFromDevice();
+}
+
+void FSpatialGDKEditorCommandLineArgsManager::OnLauncherFinished(bool bSuccess, double ExecutionTime, int32 ReturnCode)
+{
+	RemoveCommandLineFromDevice();
+}
+
+void FSpatialGDKEditorCommandLineArgsManager::RemoveCommandLineFromDevice()
+{
+	if (bAndroidDevice)
+	{
+		RemoveCommandLineFromAndroidDevice();
+	}
+}
+
+void FSpatialGDKEditorCommandLineArgsManager::OnLaunch(ILauncherWorkerPtr LauncherWorkerPtr, ILauncherProfileRef LauncherProfileRef)
+{
+	LauncherWorkerPtr->OnCanceled().AddRaw(this, &FSpatialGDKEditorCommandLineArgsManager::OnLauncherCanceled);
```

Comment:
I remember commenting on this in the original PR, but don't remember the response. we should unbind this when needed or only add it once. Right now we will add it with each launch, right?

Reply:
Each launch creates a new instance of `FLauncherWorker` so we don't need to unbind the callback.